[fix] Update apt-cache before installing dependencies

obitech opened the pull request:
Updates the apt cache before installing Grafana dependencies
through the package manager. This resolves a bug where old dependencies,
or those with a changed name, would not be installed due to
the package manager not being able to find them.

Fixes #200 
Ref #201

obitech commented:
Hmm it seems to fail at the pre-validate stage but I have trouble parsing the output. Any pointers?

paulfantom commented:
The issue is with molecule itself. More in https://github.com/cloudalchemy/skeleton/pull/5

obitech commented:
I've migrated the changes but now there are a lot of warnings and errors from `yamlint` in different files. Let's see what CI says

paulfantom commented:
Seems like migration to molecule v3 might take more time than I though :(

Can you revert previous change and just pin molecule to version v2 in `test-requirements.txt`?
This would need just a change of one line to be like this one:
```
molecule>=2.15.0,<3.0.0
```

Sorry for those problems. I'll try to tackle it during weekend.

obitech commented:
I've reverted and squashed commits.

> Sorry for those problems. I'll try to tackle it during weekend.

No worries! Let me know if I can help with anything 🙂

paulfantom commented:
Seems like tests don't like refreshing apt cache:
> ERROR: Idempotence test failed because of the following tasks:
> * [xenial] => ansible-grafana : Update apt cache
> * [bionic] => ansible-grafana : Update apt cache

from https://travis-ci.com/cloudalchemy/ansible-grafana/jobs/291642777#L907

If this is necessary we might need to use `apt` module for installing grafana on debiand-based or apt-based systems and `package` module on others. However IMHO this is very "dirty" solution and should be avoided if possible.

obitech commented:
> If this is necessary we might need to use apt module for installing grafana on debiand-based or apt-based systems and package module on others. However IMHO this is very "dirty" solution and should be avoided if possible.

I agree, this is not something I'd want to do either.

> Maybe we can move refreshing cache to block in which we set up GPG keys for repository?

If you mean the block later in the task, I don't think it works because `apt cache` needs to be refreshed before the dependencies are installed, right at the beginning. I've tried encapsulating the whole `Install dependencies` tasks as a block, let's see if that works.
